service/githubhook/githubhook.go

```diff
@@ -0,0 +1,100 @@
+// Package githubhook implements handling and verification of github webhooks.
+package githubhook
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	signaturePrefix = "sha1="
+	signatureLength = 45 // len(SignaturePrefix) + len(hex(sha1))
+)
+
+// Hook is an inbound github webhook.
+type Hook struct {
+
+	// Id specifies the Id of a github webhook request.
+	Id string
+	// Event contains unmarshaled webhook.
+	Event Event
+	// Event specifies the event name of a github webhook request.
+	EventName string
+	// Signature specifies the signature of a github webhook request.
+	Signature string
+	// Payload contains the raw contents of the webhook request.
+	Payload []byte
+}
+
```

Comment:
We structure code like this (this is also upstream convention):

- Type definition
- Constructors (New and Parse in this case)
- Methods (SignedBy in this case)
- Helper functions (signBody in this case)

Same applies to `Oncall`.